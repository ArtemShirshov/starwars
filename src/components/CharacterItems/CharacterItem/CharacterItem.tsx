import React, { FC } from 'react';
import {
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Typography,
} from '@mui/material';

import { CharacterType } from 'store/CharacterItems';

type CharacterItemType = Pick<CharacterType, 'name' | 'id'>;

export const CharacterItem: FC<CharacterItemType> = ({ name, id }) => {
  return (
    <Card
      sx={{
        height: '100%',
      }}
    >
      <CardActionArea
        sx={{
          height: '100%',
        }}
      >
        <CardMedia
          alt="green iguana"
          component="img"
          height="400"
          image={`/assets/characters/${id}.jpg`}
        />
        <CardContent>
          <Typography
            color="primary"
            gutterBottom
            sx={{
              margin: 0,
            }}
            variant="h6"
          >
            {name}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};
