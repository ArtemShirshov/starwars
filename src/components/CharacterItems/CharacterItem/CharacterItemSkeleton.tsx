import { Card, CardActionArea, CardContent, Skeleton } from '@mui/material';
import React from 'react';

export const CharacterItemSkeleton = () => {
  return (
    <Card>
      <CardActionArea>
        <Skeleton
          animation="wave"
          sx={{
            height: 400,
          }}
          variant="rectangular"
        />
        <CardContent>
          <Skeleton animation="wave" height={32} width="80%" />
        </CardContent>
      </CardActionArea>
    </Card>
  );
};
