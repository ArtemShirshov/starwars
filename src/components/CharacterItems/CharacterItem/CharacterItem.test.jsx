import React from 'react';
import renderer from 'react-test-renderer';

import { CharacterItem } from './CharacterItem';

jest.mock('@mui/material', () => ({
  ...jest.requireActual('@mui/material'),
  Card: 'Card',
  CardContent: 'CardContent',
  CardMedia: 'CardMedia',
  Typography: 'Typography',
  CardActionArea: 'CardActionArea',
}));

describe('CharacterItems', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CharacterItem id={1} name="Luke" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
