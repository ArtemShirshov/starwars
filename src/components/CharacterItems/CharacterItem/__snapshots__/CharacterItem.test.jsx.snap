// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItems renders correctly 1`] = `
<Card
  sx={
    Object {
      "height": "100%",
    }
  }
>
  <CardActionArea
    sx={
      Object {
        "height": "100%",
      }
    }
  >
    <CardMedia
      alt="green iguana"
      component="img"
      height="400"
      image="/assets/characters/1.jpg"
    />
    <CardContent>
      <Typography
        color="primary"
        gutterBottom={true}
        sx={
          Object {
            "margin": 0,
          }
        }
        variant="h6"
      >
        Luke
      </Typography>
    </CardContent>
  </CardActionArea>
</Card>
`;
